pytorch_lightning/metrics/functional/image_gradients.py

```diff
@@ -0,0 +1,73 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Tuple
+
+import torch
+
+
+def _image_gradients_validate(img: torch.Tensor) -> torch.Tensor:
+
+    if not isinstance(img, torch.Tensor):
+        raise TypeError(f"`img` expects a value of <torch.Tensor> type but got {type(img)}")
+    if img.ndim != 3 and img.ndim != 4:
+        raise RuntimeError(f"`img` expects a 3D or 4D tensor but got {img.ndim}D tensor")
+
+
+def _compute_image_gradients(img: torch.Tensor) -> Tuple[torch.Tensor, torch.Tensor]:
+
+    batch_size, channels, height, width = img.shape
+
+    dy = img[..., 1:, :] - img[..., :-1, :]
+    dx = img[..., :, 1:] - img[..., :, :-1]
+
+    shapey = [batch_size, channels, 1, width]
+    dy = torch.cat([dy, torch.zeros(shapey, device=img.device, dtype=img.dtype)], dim=2)
+    dy = dy.view(img.shape)
+
+    shapex = [batch_size, channels, height, 1]
+    dx = torch.cat([dx, torch.zeros(shapex, device=img.device, dtype=img.dtype)], dim=3)
+    dx = dx.view(img.shape)
+
+    return dy, dx
+
+
+def image_gradients(img: torch.Tensor) -> Tuple[torch.Tensor, torch.Tensor]:
+    """
+    Computes the gradients of a given Image
+
+    Args:
+        img: input image tensor
```

Comment:
Please add description on the expected shape of this tensor. How will a 3D tensor be converted to a 4D tensor (is it the batch or channel dimension that will be unsqueezed)

Reply:
resolved in the latest commit